osbrain/tests/test_agent_sync_publications.py

```diff
@@ -73,6 +73,164 @@ def on_error(agent):
     agent.error_log.append('error')
 
 
+@pytest.mark.parametrize('socket_type', ['PUB', 'SYNC_PUB'])
+@pytest.mark.parametrize('handler', [append_received, {b'': append_received}])
+def test_unsubscribe(nsproxy, socket_type, handler):
+    '''
+    Test unsubscribing from topics works fine.
+    '''
+    server = run_agent('server')
+    client = run_agent('client')
+
+    addr = server.bind(socket_type, alias='pub', handler=lambda: None)
+    client.set_attr(received=[])
+    client.connect(addr, alias='sub', handler=handler)
+
+    # Stablish a connection
+    server.each(0.1, 'send', 'pub', 'connecting...')
+    assert wait_agent_attr(client, data='connecting...')
+
+    # Clear subscription to default '' topic (do not receive anything)
+    client.unsubscribe('sub', '')
+
+    # Message not received
+    server.send('pub', 'not received')
+    assert not wait_agent_attr(client, data='not received',
+                               timeout=0.5)
```

Comment:
There might be a sync problem in this test I think: maybe we can send from the publisher before the client completed the unsubscription?

Maybe we could use the incremental numbers trick? Publish two topics `a`/`b`, then unsubscribe from one of them. Then wait for the last received value difference (`a` - `b`) to be greater than 1?

Maybe we can create a `wait_for` function, which accepts a function to execute periodically on the agent until it succeeds or the timeout is reached.